### PR TITLE
BoardConfig.mk: Update userdata partition size from fastboot getvar

### PR DESCRIPTION
**Pull request by Haxk20**
XQ-CT54 reports:
    fastboot getvar all 2>&1 | rg "size:(user)"
    (bootloader) partition-size:userdata: 0x36AEB3F000

So lets update the value (accounting for encryption as well)

Also included removal of leading zero in boot, DTBO part sizes as they will be ignored regardless.

### CODE REVIEW

File: BoardConfig.mk [hunk 1]
```diff
@@ -31,8 +31,8 @@ BOARD_KERNEL_CMDLINE += androidboot.fstab_suffix=pdx223
 
 # Partition information
 BOARD_FLASH_BLOCK_SIZE := 131072 # (BOARD_KERNEL_PAGESIZE * 64)
-BOARD_BOOTIMAGE_PARTITION_SIZE := 0x06000000
-BOARD_DTBOIMG_PARTITION_SIZE := 0x01800000
+BOARD_BOOTIMAGE_PARTITION_SIZE := 0x6000000
+BOARD_DTBOIMG_PARTITION_SIZE := 0x1800000
```
Comment (MarijnS95) [inline BoardConfig.mk:35]: Drive-by?

Comment (Haxk20) [reply BoardConfig.mk:35]: Just removal of leading zero.

Comment (jerpelea) [reply BoardConfig.mk:35]: please update only the data partition!
